src/librbd/io/ImageRequestWQ.cc

```diff
@@ -597,6 +597,12 @@ void *ImageRequestWQ<I>::_void_dequeue() {
     return nullptr;
   }
 
+  if (write_op) {
+    m_image_ctx.write_iops_throttle->get(1);
```

Comment:
This will block the thread pool's worker thread. These need to be asynchronous callbacks when throttling is required. Also, if I set me write qos to X and my read to Y (where Y > X), all my reads can be throttled if I go past my write limit.  Perhaps just start w/ a single ```rbd_iops_max``` configuration parameter that throttles both -- or provide separate read and write queues for throttled IOs.